### DISCUSSION TITLE
Mailer > Office 365 EOL Basic Auth, Moder Auth?

### DISCUSSION THREAD

**Discussion by ucola** [symfony/symfony#46732] [Q&A] [answered]
Hi all

We use in some cases SMTP for sending mails with Office365.
Microsoft announced, that SMTP with Basic Auth will be EOL and they will be replaced with Moder Authentication.

Does anyone know when Moder will be available?

Thank you for your reply

**Comment by stof** [comment, marked as answer]
When someone sends a PR to implement it.

**Comment by ucola** [comment]
Hi @stof 
thank you for your reply... sorry, I have never create a PR. Can you explain me where I can do this?

**Reply by stof** [reply]
See https://symfony.com/doc/current/contributing/code/index.html for the doc about how to contribute

**Reply by ucola** [reply]
Sorry but I don't find any about PR, only how to contribute myself... Would be nice, if you can told me where I can send a PR.

**Reply by stof** [reply]
Doing a pull request is the way you submit a contribution.

**Reply by zippytiff** [reply]
Hi,

I have just hit this issue too

Is Modern Authentication going to be possible, or do i need to look for a new solution

Thanks

ZT

**Reply by stof** [reply]
my answer from June still apply. This will be possible once someone contributes to implement it.

**Reply by zippytiff** [reply]
Hi, do you mean money or code ?

ZT

**Reply by stof** [reply]
I mean code. But if you have money instead, you can hire someone to do that code.

**Comment by gordinskiy** [comment]
If someone like me was trying to figure out how to solve this problem - most likely you should not:
1) There are no plans to disable Basic Auth for office365 SMTP:
> We are not making any changes to SMTP Auth starting on October 1st. If it works today, it'll keep working from October 1st, until the day when SMTP Auth support for basic auth goes away (which has not been announced).

https://techcommunity.microsoft.com/t5/exchange-team-blog/basic-authentication-deprecation-in-exchange-online-september/bc-p/3619536/highlight/true#M34002


2) There are no scopes that allow you to send email using Modern Authentication in background Application.

> Note As per the current test with SMTP Oauth 2.0 client credential flow with non-interactive sign in is not supported. 
Moder Authentication

https://learn.microsoft.com/en-us/exchange/client-developer/legacy-protocols/how-to-authenticate-an-imap-pop-smtp-application-by-using-oauth#smtp-protocol-exchange

Instructions on how to enable  SMTP AUTH for new applications:
https://learn.microsoft.com/en-us/exchange/clients-and-mobile-in-exchange-online/authenticated-client-smtp-submission#use-the-microsoft-365-admin-center-to-enable-or-disable-smtp-auth-on-specific-mailboxes

@zippytiff Perhaps this will be useful to you

**Reply by zippytiff** [reply]
Thats amazing, thank you, ill review

Regards

ZT
